docker/Dockerfile

```diff
@@ -71,11 +71,17 @@ RUN --mount=type=cache,target="/root/.cache/pip",sharing=locked \
 
 HEALTHCHECK --interval=5s --timeout=5s --start-period=5s --retries=1 CMD curl --fail http://localhost:8080/health/ || exit 1
 
+ARG PYTHON_VER
 # Generate nautobot user and its required dirs for later consumption
 RUN mkdir /opt/nautobot /opt/nautobot/.cache /prom_cache /source && \
     groupadd --gid 999 --system nautobot && \
     useradd --uid 999 --gid 999 --system --shell /bin/bash --create-home --home-dir /opt/nautobot nautobot && \
-    chown -R nautobot:nautobot /opt/nautobot /prom_cache /source
+    chown -R nautobot:nautobot \
+    /opt/nautobot \
+    /prom_cache \
+    /source \
+    /usr/local/bin \
+    /usr/local/lib/python${PYTHON_VER}/site-packages
```

Comment:
This seems like a security risk allowing the execution user to write to `site-packages` and bin, the point of not running as root is to not allow this specifically

Reply:
@snaselj @nniehoff Can/should we run the `pip` or `poetry` commands as `root` then?

Reply:
There are another places where user `nautobot` has write access in `final` image, should I fix it as well?

```
COPY --from=build-nautobot --chown=nautobot:nautobot /source/dist /source/dist
COPY --from=final-dev --chown=nautobot:nautobot /opt/nautobot/nautobot_config.py /opt/nautobot/nautobot_config.py
COPY --from=final-dev --chown=nautobot:nautobot /opt/nautobot/ui /opt/nautobot/ui
COPY --chown=nautobot:nautobot docker/uwsgi.ini /opt/nautobot
```

Reply:
I think this is a slippery slope: by @nniehoff’s comments none of this should be writable but we need to create the docs and UI so maybe that needs to be run as root too. However everything starts with the site-packages so let’s start with the Python executables not being writable.

If this is permitted in the Venv install method, why not here?

Reply:
> @snaselj @nniehoff Can/should we run the `pip` or `poetry` commands as `root` then?

During the build phase yes they should I would just use the `USER` directive to switch back and forth

Reply:
All pip and poetry commands are run as root now.